### ISSUE TITLE
Ifpack: Remove Build Warnings

### ISSUE THREAD

**Issue by ZUUL42** [trilinos/Trilinos#4508]
@trilinos/ifpack 

## Expectations

Ifpack builds should not emit any warnings that will be promoted to errors once Werror is set in the GCC 7.2.0 automated build.
## Current Behavior

Currently Ifpack has a number of warnings that need to be handled before we can set Werror for all packages.
A recent test build was performed with -Werror set for Ifpack. The CDash report can be found [here](https://testing-vm.sandia.gov/cdash/viewBuildError.php?buildid=4623061).
## Motivation and Context

Issue #3178 is working toward turning Warnings as Errors on for all packages.

**Comment by mhoemmen** [comment]
This is one of those "`assert` only does anything in a debug build" issues.

**Comment by ZUUL42** [comment]
Thanks

**Comment by mhoemmen** [comment]
@ZUUL42 Generally I've been waiting to close issues until they merge into master (which usually autocloses the issue), but we haven't been consistent about that.  I would like us to do what Kokkos does, namely have an "in develop" tag that says "the issue was fixed in develop."

**Comment by ZUUL42** [comment]
Ahh. OK. Since the larger Werror issue broke down into so many more package level issues, I thought it good to close them as each part has been completed to help track progress more clearly.
But I can follow that practice, leaving them open until merged into master, moving forward. I can also repoen this, if your prefer. I took the "In Develop" tag as short for "in development", not "in the develop branch". Still picking up a few details here & there.

**Comment by mhoemmen** [comment]
@ZUUL42 I'm not sure how Trilinos views that tag -- Kokkos interprets it as "fix exists in the develop branch."  I would like Trilinos to have a tag that expresses that idea.  But no worries :-)
